cli/config/credentials/default_store.go

```diff
@@ -2,21 +2,70 @@ package credentials
 
 import "os/exec"
 
-// DetectDefaultStore return the default credentials store for the platform if
-// no user-defined store is passed, and the store executable is available.
-func DetectDefaultStore(store string) string {
-	if store != "" {
+// DetectDefaultStore returns the credentials store to use. If no user-defined
+// custom helper is passed.
+//
+// Some platforms define a preferred helper, in which case it attempts to look
+// up the helper binary before falling back to the platform's default.
+//
+// If no user-defined helper is passed, and no helper is found, it returns an
+// empty string, which means credentials are stored unencrypted in the CLI's
+// config-file without the use of a credentials store.
+func DetectDefaultStore(customStore string) string {
+	if customStore != "" {
 		// use user-defined
-		return store
+		return customStore
 	}
+	if preferred := findPreferredHelper(); preferred != "" {
+		return preferred
+	}
+	if defaultHelper == "" {
+		return ""
+	}
+	if _, err := exec.LookPath(remoteCredentialsPrefix + defaultHelper); err != nil {
+		return ""
+	}
+	return defaultHelper
+}
+
+// overridePreferred is used to override the preferred helper in tests.
+var overridePreferred string
 
-	platformDefault := defaultCredentialsStore()
-	if platformDefault == "" {
+// findPreferredHelper detects whether the preferred credentials-store and
+// its helper binaries are installed. It returns the name of the preferred
+// store if found, otherwise returns an empty string to fall back to resolving
+// the default helper.
+//
+// Note that the logic below is currently specific to detection needed for the
+// "pass" credentials-helper on Linux (which is the only platform with a preferred
+// helper). It is put in a non-platform specific file to allow running tests
+// on other platforms.
+func findPreferredHelper() string {
+	preferred := preferredHelper
+	if overridePreferred != "" {
+		preferred = overridePreferred
+	}
+	if preferred == "" {
+		return ""
+	}
+
+	// Note that the logic below is specific to detection needed for the
+	// "pass" credentials-helper on Linux (which is the only platform with
+	// a "preferred" and "default" helper. This logic may change if a similar
+	// order of preference is needed on other platforms.
```

Comment:
I realize you moved this here in order to make this easier to test cross-platform, but this is exactly the type of thing that would make sense to put in a `_linux.go` file and not leak to the rest of the code – otherwise, we need to have it here with a number of comments explaining why, while it's unnecessary for anyone worrying about any other platform to look at).